### DISCUSSION TITLE
Create New App Error - If internet is not on available

### DISCUSSION THREAD

**Discussion by Manishkparmar80** [Budibase/budibase#1842] [Help] [answered]
it is observed that while creating new app, Budibase is trying to contact URL https://registry.npmjs.org/@budibase/standard-components/-/standard-components-0.9.64.tgz 

If internet connection is not available then it display error. attached snap of error.
![Error](https://user-images.githubusercontent.com/84949590/123535726-47a97280-d743-11eb-86d4-7de1c571d4f4.jpg)

**Comment by shogunpurple** [comment, marked as answer]
@Manishkparmar80 budibase pulls down the latest component library from NPM when you create a new app. It requires internet access for this step

**Reply by Manishkparmar80** [reply]
@shogunpurple  OK. noted. Is it only create new app time it pulls component library or it do it frequently?

**Reply by shogunpurple** [reply]
Yep, it's only on app creation

**Comment by Manishkparmar80** [comment]
@shogunpurple Also confirm internet is required on PC on which docker is hosted or internet is required on the PC from which new application is creating ?

**Reply by shogunpurple** [reply]
@Manishkparmar80 internet is required on the server that budibase is hosted on, so the machine running docker. We may look at potentially avoiding this requirement in future, but right now that's the case.

**Reply by Manishkparmar80** [reply]
@shogunpurple  budibase is only pulls down the latest component library from NPM when you create a new app.  With this concept,  existing application's component library is not getting update and i am not able to see updated component into existing applications.   How to i can update latest component library for existing app.

**Reply by shogunpurple** [reply]
@Manishkparmar80 at the moment the only way to do this is to export your app and import it again. We will be implementing an upgrade path for apps in the current UI over the next few weeks
